src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -75,6 +79,97 @@ public NetworkModificationService(NetworkStoreService networkStoreService, Netwo
+                .flatMapIterable(network -> doModification(network, networkUuid, () -> {
+                    if (disconnectLineBothSides(network, lineId)) {
+                        network.getLine(lineId).newExtension(BranchStatusAdder.class).withStatus(BranchStatus.Status.FORCED_OUTAGE).add();
+                    } else {
+                        throw new NetworkModificationException(MODIFICATION_ERROR, "Unable to disconnect both line ends");
+                    }
+                }
+                ));
+    }
+
+    public Flux<ElementaryModificationInfos> energiseLineEnd(UUID networkUuid, String lineId, String side) {
+        return getNetwork(networkUuid)
+                .filter(network -> network.getLine(lineId) != null)
+                .switchIfEmpty(Mono.error(new NetworkModificationException(LINE_NOT_FOUND, lineId)))
+                .flatMapIterable(network -> doModification(network, networkUuid, () -> {
+                    Terminal terminalToConnect = network.getLine(lineId).getTerminal(Branch.Side.valueOf(side));
+                    boolean isTerminalToConnectConnected = terminalToConnect.isConnected();
+                    if (!isTerminalToConnectConnected) {
+                        isTerminalToConnectConnected = terminalToConnect.connect();
+                    }
+                    Terminal terminalToDisconnect = network.getLine(lineId).getTerminal(Branch.Side.valueOf(side) == Branch.Side.ONE ? Branch.Side.TWO : Branch.Side.ONE);
+                    boolean isTerminalToDisconnectDisconnected = !terminalToDisconnect.isConnected();
+                    if (!isTerminalToDisconnectDisconnected) {
+                        isTerminalToDisconnectDisconnected = terminalToDisconnect.disconnect();
+                    }
+                    if (isTerminalToConnectConnected && isTerminalToDisconnectDisconnected) {
+                        network.getLine(lineId).newExtension(BranchStatusAdder.class).withStatus(BranchStatus.Status.IN_OPERATION).add();
+                    } else {
+                        throw new NetworkModificationException(MODIFICATION_ERROR, "Unable to energise line end");
+                    }
+
+                }
+                ));
+    }
+
+    public Flux<ElementaryModificationInfos> switchOnLine(UUID networkUuid, String lineId) {
+        return getNetwork(networkUuid)
+                .filter(network -> network.getLine(lineId) != null)
+                .switchIfEmpty(Mono.error(new NetworkModificationException(LINE_NOT_FOUND, lineId)))
+                .flatMapIterable(network -> doModification(network, networkUuid, () -> {
+                    Terminal terminal1 = network.getLine(lineId).getTerminal1();
+                    boolean terminal1Connected = terminal1.isConnected();
+                    if (!terminal1Connected) {
+                        terminal1Connected = terminal1.connect();
+                    }
```

Comment:
boolean terminal1Connected = terminal1.isConnected() || terminal1.connect();

Reply:
Modified

Reply:
Done ?

Reply:
No, not done here, I made a mistake. Now it's done !